src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -342,54 +323,11 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             traceError(`Failed to create notebook controller for ${kernelConnection.id}`, ex);
         }
     }
-    /**
-     * In our tests, preferred controllers are setup as the active controller.
-     *
-     * This method is called on when running tests, else in the real world,
-     * users need to select a kernel (preferred is on top of the list).
-     */
-    private async setAsActiveControllerForTests(controller: VSCodeNotebookController, notebook: NotebookDocument) {
```

Comment:
Moved into controller.
The controller was calling this method via a callback. I.e. manager creates controller, controller calls a method in manager via a callback, yuck...